### ISSUE TITLE
Certificated not taken into account for combined status 

### ISSUE THREAD

**Issue by svenfoo** [corona-warn-app/cwa-app-android#4742]
## Technical details

- Device name: Motorola One
- Android version: 10
- App version: 2.16.2

## Describe the bug

In the certificates view the app displays a combined status. With three vaccinations imported for me into the app, my status is **2G**. Now when I get myself tested and import the test result into the app I expect the status to change to **2G+**. This is however not happening, the status remains at **2G**.

## Steps to reproduce the issue

1. Go to 'Certificates' for the owner of this app.
2. The status is displayed as **2G**.
3. Import a test result for the owner of the app. Name and birth date are identical.
4. The status is still displayed as **2G**.
-->

## Expected behaviour

The status is still displayed as **2G+**.

## Additional context

I have seen this feature working correctly on other people's devices. The only difference I found in the environment is that I am using my Android with *English* as system language. Might the problem be related to language settings?

## Possible fix

In the certificates my *birth date* is displayed as **Born 09/14/1970**. On the test result it's **Born Sep 14, 1970**. As far as I understand a test result is only matched to the certificates if name and birth date match. I haven't looked at the implementation, but I could imagine that the date comparison might somehow be affected by language settings.

**Comment by dsarkar** [comment]
Hi @svenfoo, 

thanks for the report. Is the test certificate shown/grouped together with the other certificates?
See https://www.coronawarn.app/assets/screenshots/2-16/en/android/PersonDetailsFragment_cwa_2.png

Is your standardised name exactly written the same way on all certificates? 
<img width="283" alt="image" src="https://user-images.githubusercontent.com/1409741/151195353-08e77f6c-9dbb-40b8-805c-ee842ec0fd81.png">

Thanks for your feedback.

Best wishes, DS

---
Corona-Warn-App Open Source Team

**Comment by Shineson1001** [comment]
Hi,
i think I have the same problem.

Three vaccinations.
![grafik](https://user-images.githubusercontent.com/22273771/151232615-265b7174-a887-4a30-aa62-da3821115e41.png)

A recent Test.
![grafik](https://user-images.githubusercontent.com/22273771/151232696-84a34a74-5e9a-4958-9a81-1131a6b58d31.png)

![grafik](https://user-images.githubusercontent.com/22273771/151232661-5c8d7af5-6048-46b1-a394-d5926c5e98df.png)

The app shows "2G" only
![grafik](https://user-images.githubusercontent.com/22273771/151232751-cd07b5b6-da46-4964-ae08-317b66ccc2dd.png)

**Comment by Ein-Tim** [comment]
@svenfoo @Shineson1001 

Did you actually get a certificate for your test or is the result only shown on the "Status" tab?

**Comment by Shineson1001** [comment]
Hi,
no, i did not get a certificate. It is only shown in the "Status" tab.

**Comment by Ein-Tim** [comment]
@Shineson1001

This does not count for the 2G+-Status, only test certificates are taken into account.

**Comment by Shineson1001** [comment]
Hi @Ein-Tim,
ok, i my oppion the app should display "2G+"
If you you have 2 vaccinations and a negativ corona test .. it is "2G+"
and 3 vaccinations and a negativ corona test .. it is "2G++" :-)

**Comment by Ein-Tim** [comment]
@Shineson1001 Your test result does not seem to be personalized, thus the app can't know that this is actually your test and not from another person.

**Comment by Shineson1001** [comment]
Hi,
if I make an appointment for a test, I use that option:
![grafik](https://user-images.githubusercontent.com/22273771/151238680-af423d85-a906-4c8d-a03e-16e44097bce6.png)
![grafik](https://user-images.githubusercontent.com/22273771/151238805-831a655f-e440-4a36-89fc-9a5d05614f6b.png)



After that, you get an QR-Code for the CWA

**Comment by Ein-Tim** [comment]
@Shineson1001 

Yes as you can see it says "nicht-namentliche Anzeige" thus the app does not know from whom the test is and also can't request a test certificate.

**Comment by Shineson1001** [comment]
ok. thanks.
